include/unifex/with_scheduler_affinity.hpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/await_transform.hpp>
+#include <unifex/connect_awaitable.hpp>
+#include <unifex/finally.hpp>
+#include <unifex/scheduler_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/type_traits.hpp>
+#include <unifex/unstoppable.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _wsa {
+
+template <typename Sender, typename Scheduler>
+struct _wsa_sender_wrapper final {
+  class type;
+};
+
+template <typename Sender, typename Scheduler>
+static auto
+_make_sender(Sender&& sender, Scheduler&& scheduler) noexcept(noexcept(finally(
+    static_cast<Sender&&>(sender),
+    unstoppable(schedule(static_cast<Scheduler&&>(scheduler)))))) {
+  return finally(
+      static_cast<Sender&&>(sender),
+      unstoppable(schedule(static_cast<Scheduler&&>(scheduler))));
+}
+
+template <typename Sender, typename Scheduler>
+using wsa_sender_wrapper =
+    typename _wsa_sender_wrapper<Sender, Scheduler>::type;
```

Comment:
nit: just `_wsa_sender`?

Reply:
Maybe next time I'm in here; I'd like to get this merged so we can sync it internally.